### ISSUE TITLE
pd.concat() duplicates the data

### ISSUE THREAD

**Issue by just-sabyr** [fireducks-dev/fireducks#19]
`Python
# Save original to avoid having to rerun the whole notebook
X = original_X
X_test = original_X_test

# Initialize OHE
ordinal_encoder = OrdinalEncoder(handle_unknown='use_encoded_value', unknown_value=-1)
ohe = OneHotEncoder(sparse_output=False)

# Apply Ordinal Encoding to non-categorical features 
X[non_categorical] = ordinal_encoder.fit_transform(X[non_categorical])
X_test[non_categorical] = ordinal_encoder.transform(X_test[non_categorical])

# Transform categorical features
X_array = ohe.fit_transform(X[categorical])
X_test_array = ohe.transform(X_test[categorical])
print('X_test_arraylen: ', len(X_test_array))

ohe_features = ohe.get_feature_names_out(categorical)
X_ohe = pd.DataFrame(X_array, columns=ohe_features, index=X.index)
X_test_ohe = pd.DataFrame(X_test_array, columns=ohe_features, index=X_test.index)

print('len(X_test_ohe): ', len(X_test_ohe))

# Drop original categorical columns before concatenation
X = X.drop(columns=categorical)
X_test = X_test.drop(columns=categorical)

print('len(X_test): ', len(X_test_categorical))

X = pd.concat([X, X_ohe], axis=1)
X_test = pd.concat([X_test, X_test_ohe], axis=1)

print('len(X_test): ', len(X_test))

assert X.columns.equals(X_test.columns), "There is no helping it, reconsider everything"
assert len(X_test) == len(test_df), "Yeah, X_test is probably not concatenated well"
`
The line X_test = pd.concat([X_test, X_test_ohe], axis=1) is resulting in duplication of X_test and the len(X_test) became twice more

**Comment by qsourav** [comment]
Hey @just-sabyr ,

Thank you for reporting the issue.

When I tried the above code with a randomly generated sample data, it executed well as expected and the issue couldn't be reproduced. Please consider attached the sample program that I tried.

Can you provide a little more information or some hints related to data to reproduce the problem?

**Comment by just-sabyr** [comment]
@qsourav, sorry for the late reply. Thanks for taking your time. I had solved this issue by just using the regular pandas the actual code is https://www.kaggle.com/code/sabyrbazarymbetov/used-car-price-eda

**Comment by qsourav** [comment]
Hi @just-sabyr,

Thank you for the update. Although I didnot understand the root-cause of the issue, but when I tried your kaggle code with FireDucks, it worked without any issue: https://www.kaggle.com/code/qsourav91/used-car-price-eda-with-fireducks/edit
